tests/test_integration/test_frame_extraction.py

```diff
@@ -1,51 +1,490 @@
+
+    We modify the defaults with values that are more convenient for testing.
+
+    Parameters
+    ----------
+    cli_inputs_dict : dict
+        a dictionary with parameters for the frame extraction
+
+    Returns
+    -------
+    typer.main.Typer
+        an app with the same functionality as the one defined for
+        `compute_and_extract_frames_to_label` but with different default values.
+    """
+    from crabs.bboxes_labelling.extract_frames_to_label_w_sleap import (
+        compute_and_extract_frames_to_label,
+    )
+
+    # instantiate app
+    app = typer.Typer(rich_markup_mode="rich")
+
+    # link mocked command to app
+    # change the defaults so that they match cli_inputs_dict
+    @app.command()
+    def mock_combine_and_format_annotations(
+        list_video_locations: list[str],
+        output_path: str = cli_inputs_dict["output-path"],
+        output_subdir: Optional[str] = None,
+        video_extensions: list[str] = cli_inputs_dict["video-extensions"],
+        initial_samples: int = cli_inputs_dict["initial-samples"],
+        sample_method: str = "stride",
+        scale: float = cli_inputs_dict["scale"],
+        feature_type: str = "raw",
+        n_components: int = cli_inputs_dict["n-components"],
+        n_clusters: int = cli_inputs_dict["n-clusters"],
+        per_cluster: int = cli_inputs_dict["per-cluster"],
+        compute_features_per_video: bool = True,
+    ):
+        return compute_and_extract_frames_to_label(
             list_video_locations,
-            list_user_extensions,
+            output_path=output_path,
+            output_subdir=output_subdir,
+            video_extensions=video_extensions,
+            initial_samples=initial_samples,
+            sample_method=sample_method,
+            scale=scale,
+            feature_type=feature_type,
+            n_components=n_components,
+            n_clusters=n_clusters,
+            per_cluster=per_cluster,
+            compute_features_per_video=compute_features_per_video,
         )
 
-        # check list of SLEAP videos matches the list of files
-        assert len(list_sleap_videos) == len(list_files)
-        assert len(list_sleap_videos) == len(list_files)
+    return app
+
+
+def list_files_in_dir(input_dir: str) -> list:
+    """Lists files in input directory
+
+    Parameters
+    ----------
+    input_dir : str
+        path to directory
+
+    Returns
+    -------
+    list
+        list of files in input directory
+    """
+
+    return [
+        f
+        for f in Path(input_dir).glob("*")
+        if f.is_file() and not f.name.startswith(".")
+    ]
+
+
+def check_output_files(list_input_videos: list, cli_dict: dict):
+    """Run assertions on output files from frame extraction
+
+    Parameters
+    ----------
+    list_input_videos : list
+        list of videos used for frame extraction
+    cli_dict : dict
+        a dictionary with the parameters of the frame extraction
```

Comment:
Incidentally (this is a bit bike-sheddy) I don't mind not having a pramaters list for the test functions if they're fixtures. 

It's not like the test functions are public or form part of your API or anything.

... also this file is quite long as it is.